packages/contractkit/src/wallets/signers/rpc-signer.ts

```diff
@@ -0,0 +1,105 @@
+import { ensureLeading0x, normalizeAddressWith0x, trimLeading0x } from '@celo/utils/src/address'
+import BigNumber from 'bignumber.js'
+import BN from 'bn.js'
+import { Tx } from 'web3-core'
+import { JsonRpcResponse } from 'web3-core-helpers'
+import { RpcCaller } from '../../utils/rpc-caller'
+import { decodeSig } from '../../utils/signing-utils'
+import { Signer } from './signer'
+
+const currentTimeInSeconds = () => Math.round(Date.now() / 1000)
+
+const toRpcHex = (val: string | number | BigNumber | BN | undefined) => {
+  if (typeof val === 'number' || val instanceof BigNumber) {
+    return ensureLeading0x(val.toString(16))
+  } else if (typeof val === 'string') {
+    return ensureLeading0x(val)
+  } else {
+    return '0x0'
+  }
+}
+
+/**
+ * Implements the signer interface on top of the JSON-RPC interface.
+ */
+export class RpcSigner implements Signer {
+  /**
+   * Construct a new instance of the RPC signer
+   *
+   * @param rpc RPC caller instance
+   * @param account Account address derived from the private key to be called in init
+   * @param unlockBufferSeconds Number of seconds to shrink the unlocked duration by to account for
+   * latency and timing inconsistencies on the node
+   * @param unlockTime Timestamp in seconds when the signer was last unlocked
+   * @param unlockDuration Number of seconds that the signer was last unlocked for
+   *
+   */
+  constructor(
+    protected rpc: RpcCaller,
+    protected account: string,
+    protected unlockBufferSeconds = 5,
+    protected unlockTime = -1,
+    protected unlockDuration = -1
+  ) {}
+
+  async init(privateKey: string, passphrase: string) {
+    const response = await this.rpc.call('personal_importRawKey', [
+      trimLeading0x(privateKey),
+      passphrase,
+    ])
+    this.checkResponse(response, 'init')
+    return response.result!
+  }
+
+  async signRawTransaction(tx: Tx): Promise<any> {
+    if (normalizeAddressWith0x(tx.from! as string) !== this.account) {
```

Comment:
what's `from`'s type if not string?

Reply:
this is a product of extending web3's types
apparently a number is a valid from because in some ethereum clients it is used as an index for the account keystore
we don't use this anywhere though